recursion/gnark/sp1/groth16.go

```diff
@@ -66,66 +65,48 @@ func BuildGroth16(buildDir string) error {
 	return nil
 }
 
-// Generate a gnark groth16 proof for a given witness and write the proof to a file. Reads the
-// R1CS, the proving key and the verifier key from the build directory.
-func ProveGroth16(buildDir string, witnessPath string, proofPath string) error {
-	// Read the R1CS.
-	fmt.Println("Reading r1cs...")
-	r1csFile, err := os.Open(buildDir + "/circuit_groth16.bin")
+func ProveVerifyAndSerializeGroth16(witnessInput WitnessInput, r1cs constraint.ConstraintSystem, pk groth16.ProvingKey, vk groth16.VerifyingKey) (Groth16Proof, error) {
```

Comment:
Note: I don't feel too strongly about this being in one function, but this was the common usage pattern for Groth16 proving across the gnark server and the ffi calls.